### ISSUE TITLE
store, electron, page refactoring

### ISSUE THREAD

**Issue by HyunmoAhn** [AhKi/oh-my-desk#112]
# Structrue
```
// folder structure
app/
ㄴ store/
ㄴ process/
ㄴ render/
ㄴ constants/
```

## Store(store)
* Store is role of model.
* Use Redux pattern
* Has widget information, program config and etc.

### Why do separate store folder in root directory?
**Unify Model in electron and page.**
Electron and Page use widget information and program config. If that dates is managed separate position, It is complex and duplicate.

### Why do use `Redux` pattern
**Redux is a predictable state container**
We must use singleton pattern.(load data in disk when start app and preserve data in memory)
`Redux` is one of the singleton pattern.  
I think `Redux` is good model for managing data.
[Show about Redux](https://redux.js.org/)

### More
[How to unify the store](https://github.com/AhKi/oh-my-desk/issues/113)

## Electron(process)
Our app has `main process` and `renderer process`.
Electron processes use `Node.js` using javascript.
So, separate electron code and page(react) code.

`process` folder has code about electron like `main.js`

Main process is watching if store is changed.
If store will changed, main process run controller logic after dispatching action.   
Do you want to controll electron logic that open or focus about renderer process or another implement? Type implement code in controller using changing about store(model).
  
## Page(render)
Our app has serveral pages. Almost page is made by `react`. Some will be made by `pure js` and html.

**Page must not use electron api.**
renderer process just modify store(model) using action.
implementation of electron occur in controller of main process. 

`render` folder has code about view, page

**Comment by HyunmoAhn** [comment]
[How to unify the store](#113)

**Comment by HyunmoAhn** [comment]
Need to write document about **folder structure**.
After write document, Close this issue

**Comment by HyunmoAhn** [comment]
completed released v2.0.0
